sdk_contrib/jaeger_remote_sampler/src/main/java/io/opentelemetry/sdk/contrib/trace/jaeger/sampler/JaegerRemoteSampler.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.jaeger.sampler;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ManagedChannel;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Sampling.PerOperationSamplingStrategies;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Sampling.SamplingStrategyParameters;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.Sampling.SamplingStrategyResponse;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.SamplingManagerGrpc;
+import io.opentelemetry.exporters.jaeger.proto.api_v2.SamplingManagerGrpc.SamplingManagerBlockingStub;
+import io.opentelemetry.sdk.trace.Sampler;
+import io.opentelemetry.sdk.trace.Samplers;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/** Remote sampler that gets sampling configuration from remote Jaeger server. */
+public class JaegerRemoteSampler implements Sampler {
+  private static final Logger logger = Logger.getLogger(JaegerRemoteSampler.class.getName());
+
+  private static final int DEFAULT_POLLING_INTERVAL_MS = 60000;
+  private static final Sampler INITIAL_SAMPLER = Samplers.probability(0.001);
+
+  private final String serviceName;
+  private final SamplingManagerBlockingStub stub;
+  private Sampler sampler;
+  private final Timer pollTimer;
+
+  private JaegerRemoteSampler(
+      String serviceName, ManagedChannel channel, int pollingIntervalMs, Sampler initialSampler) {
+    this.serviceName = serviceName;
+    this.stub = SamplingManagerGrpc.newBlockingStub(channel);
+    this.sampler = initialSampler;
+
+    pollTimer = new Timer(true); // true makes this a daemon thread
```

Comment:
In general, we should prefer using the more robust ScheduledExecutorService, rather than a Timer, which can get impacted by changes to the system clock.

Reply:
Good point, I have fixed that